helix-view/src/gutter.rs

```diff
@@ -36,6 +36,32 @@ pub fn diagnostic<'doc>(
     })
 }
 
+pub fn git_diff<'doc>(
+    doc: &'doc Document,
+    _view: &View,
+    _theme: &Theme,
+    _config: &Config,
+    _is_focused: bool,
+    _width: usize,
+) -> GutterFn<'doc> {
+    Box::new(move |line: usize, _selected: bool, out: &mut String| {
+        let diff = doc.line_diffs().get(&(line + 1))?;
+
+        let (icon, style) = match diff {
+            helix_vcs::LineDiff::Added => ("▐", Style::default().fg(crate::graphics::Color::Green)),
+            helix_vcs::LineDiff::Deleted => ("▗", Style::default().fg(crate::graphics::Color::Red)),
+            helix_vcs::LineDiff::DeletedAbove => {
+                ("▝", Style::default().fg(crate::graphics::Color::Red))
+            }
+            helix_vcs::LineDiff::Modified => {
+                ("▐", Style::default().fg(crate::graphics::Color::Yellow))
+            }
```

Comment:
Fetch colors from the theme here:

https://github.com/helix-editor/helix/blob/23553bd37c5e936c022faa70bef346448a36d6a2/theme.toml#L38-L40